networkdb/networkdb_test.go

```diff
@@ -51,6 +51,9 @@ func createNetworkDBInstances(t *testing.T, num int, namePrefix string, conf *Co
 		dbs = append(dbs, db)
 	}
 
+	// Wait for all peers to be established and cluster to be created
+	time.Sleep(5 * time.Second)
```

Comment:
Is there no way to do this deterministically?   Sleep is (almost) never the right answer...

Reply:
some kind of wait make sense, as node discovery via gossip can take time. perhaps we can do a polling loop with longer expiration to ensure a) fast test time in most case b) dependable test when test machine is slow?

Reply:
Yes.   The test needs to block here, but a fixed sleep will be too long most of the time, and occasionally not long enough. :)  In the case where it's too short, we will fail messily further down the test.   It's better to poll with a timeout, then less time is wasted and a timeout is obvious.   (A reliable event from the system under test would be even better.)

Reply:
I guess you can poll for 5 seconds and break when this condition is satisfied, not a showstopper but as @euanh mentioned, best case scenario it will speed up the test
```
num == len(dbs[i].ClusterPeers())
```

Note, the sleep approach is used throughout the code, would be nice if we can refactor to start polling instead